### DISCUSSION TITLE
Version 3.4 released

### DISCUSSION THREAD

**Discussion by kjk** [sumatrapdfreader/sumatrapdf#2595] [Announcements]
I've released version 3.4.

Changes in this release:
* [Command Palette](https://www.sumatrapdfreader.org/docs/Command-Palette)
* [customizable keyboard shortcuts](https://www.sumatrapdfreader.org/docs/Customizing-keyboard-shortcuts)
* better support for epub files using mupdf's epub engine. Adds text selection and search in ebook files
* [search / translate selected text](https://www.sumatrapdfreader.org/docs/Customize-search-translation-services) with web services; we have few built-in and you can [add your own](https://www.sumatrapdfreader.org/settings/settings3-4#SelectionHandlers)
* installer: `-all-users` cmd-line arg for system-wide install
* added `Annotations.TextIconColor` and `TextIconType` advanced settings
* added `Annotations.UnderlineColor` advanced setting
* added `Annotations.DefaultAuthor` advanced setting
* `i` keyboard shortcuts inverts document colors `Shift + i` does what `i` used to do i.e. show page number
* `u` and `Shift + u` keyboard shortcuts adds underline annotation for currently selected text
* `Delete` / `Backspace` keyboard shortcuts delete an annotation under mouse cursor
support .svg files
* faster scrolling with mouse wheel when cursor over scrollbar
* add `-search` cmd-line option and `[Search("<file>", "<search-term>")]` DDE command
* a way to get list of used fonts in properties window
* support opening .heic image files (if Windows heic codec is installed)
* add experimental smooth scrolling (enabled with `SmoothScroll` advanced setting)

Also listed at https://www.sumatrapdfreader.org/docs/Version-history

Please note that to update immediately, go download from the website.

I'll enable auto-update from earlier versions once I'm convinced the new version doesn't have bugs that have to be fixed quickly.

**Comment by GitHubRulesOK** [comment]
Hmm 
3.4.0 is now previous but for any others such as 3.1.2
https://www.sumatrapdfreader.org/download-prev is currently showing blank for me

**Reply by kjk** [reply]
Fixed, thanks.

**Comment by GitHubRulesOK** [comment]
@kjk initial test for a question elsewhere

shortcut
-set-color-range #00ff00 #c0c0c0
 should change text and paper background but appears not to change text but change main background instead
MainWindowBackground = #c0c0c0
having problems applying to 3.4.1 works in 3.3.3

**Reply by kjk** [reply]
thanks, fixed

**Comment by GitHubRulesOK** [comment]
@kjk
 from 3.5.prerel, a premature comment 🙂 
>For documentation, see https://www.sumatrapdfreader.org/settings/settings3-4.html

**Comment by tony** [comment]
Is the old epub viewer (with flowing text, configurable fonts) gone forever?

```
EbookUI [
    FontName = Roboto slab
    FontSize = 13
    TextColor = #5f4b32
    BackgroundColor = #fbf0d9
    UseFixedPageUI = true
]
```

@kjk Is there an equivalent to this 3.3.x behavior in 3.4+?

**Reply by kjk** [reply]
Maybe not forever, but certainly after 3.4

**Reply by tony** [reply]
@kjk So if we want flowing text epubs, we should stay at 3.3? 

What has to be done to get it working like it was in 3.3 with `UseFixedPageUI = true`?  That mode was excellent IMO. Better than any epub reader on Windows.

Any idea how complex would be it be have flowing-text epub support again?   This sort of breaks epubs for me 😆 

I'm happy to donate $ if you can make epub reader top tier. But not sure where it is priority-wise and how much work it'd be.

**Comment by ramiropistoia** [comment]
I can't download the version 3.4.2 x64 Install. It gives me a 404 error. The rest of files doesn't give errors.

**Reply by kjk** [reply]
I can't reproduce the download error.

Try https://files.sumatrapdfreader.org/file/kjk-files/software/sumatrapdf/rel/3.4.2/SumatraPDF-3.4.2-64-install.exe and https://www.sumatrapdfreader.org/dl/rel/3.4.2/SumatraPDF-3.4.2-64-install.exe

**Reply by ramiropistoia** [reply]
thanks it's now fixed

**Comment by spec1re** [comment]
Hi, how do I a silent uninstall now?

`"%PROGRAMFILES%\SumatraPDF\SumatraPDF.exe" -uninstall`

This brings just the uninstaller up.

Thanks

**Reply by kjk** [reply]
`"%PROGRAMFILES%\SumatraPDF\SumatraPDF.exe" -uninstall -silent`

**Reply by spec1re** [reply]
Thanks.

**Comment by mdalimran-abir** [comment]
My version is showing 3.3.3.
![image](https://user-images.githubusercontent.com/68059604/171185739-73aef1cd-5b78-41a8-92b1-a432ca95a03d.png)

But when I use the "Check for Updates" option from "Help", it shows "You have the latest version."

So, do I need to manually download and install this latest version?

**Reply by kjk** [reply]
Yes, I delay the rollout of update to make sure there are not serious bugs. See https://www.sumatrapdfreader.org/docs/Update-check-doesnt-work

For now download and install from the website https://www.sumatrapdfreader.org/download-free-pdf-viewer

**Comment by biogberg** [comment]
The documentation on keyboard shortcuts (`https://www.sumatrapdfreader.org/docs/Keyboard-shortcuts`) still says the "i" should "show current page number", but the "i" key does what it should per the announcement (i.e., inverts the colors).

**Reply by kjk** [reply]
thanks, will fix
